api/circulation.py

```diff
@@ -974,7 +977,7 @@ def run(self):
                 self.log.error(
                     "%s errored out: %s", thread.api.__class__.__name__,
                     thread.exception,
-                    exc_info=thread.exception
+                    exc_info=thread.trace
```

Comment:
This lets us see a stack trace from an exception in a thread, which I think is worthwhile. It made debugging this problem a lot easier.